hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/util/NelsonData.java

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.engine.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.math3.stat.descriptive.moment.Mean;
+import org.apache.commons.math3.stat.descriptive.moment.StandardDeviation;
+import org.hawkular.alerts.api.model.condition.NelsonCondition.NelsonRule;
+import org.hawkular.alerts.api.model.data.Data;
+import org.hawkular.alerts.engine.impl.AlertProperties;
+
+/**
+ * For each dataId used in one or more NelsonConditions we have a NelsonData to track NelsonRule violations for that
+ * dataId. The NelsonData is used in the engine (Drools' working memory), it should not be exposed in the API.  The
+ * default sampleSize used to determine mean and standard deviation is 50.  It can be overridden globally via
+ * system property hawkular-alerts.nelson-rules-sample-size or environment variable HAWKULAR_NELSON_RULES_SAMPLE_SIZE.
+ * It can be overridden for a specific dataId via hawkular-alerts.nelson-rules-sample-size.<dataId> or
+ * HAWKULAR_NELSON_RULES_SAMPLE_SIZE_<dataId>.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+public class NelsonData {
+    String NELSON_RULES_SAMPLE_SIZE = "hawkular-alerts.nelson-rules-sample-size";
+    String NELSON_RULES_SAMPLE_SIZE_ENV = "HAWKULAR_NELSON_RULES_SAMPLE_SIZE";
+    String NELSON_RULES_SAMPLE_SIZE_DEFAULT = "50";
```

Comment:
I am not familiar with the Nelson rules.
One question I might have is about the population number.
I see it is configurable but I was wondering if this constant is something fixed and rarely modified (then the current impl is good) or if it is something worth to be part of the condition definition.
i.e. standard agent collects metrics per second, so this means that means are calculated per minute, perhaps letting the user to define the population size per metric/condition would be more useful (for example, for my CPU I'm interested in peaks per hour not per minute, so then for a dataId would like to increase this to a different range, like 3600).